MatrixPilot/flightplan-logo.c

```diff
@@ -443,20 +443,37 @@ void flightplan_logo_update(void)
 	}
 
 	// otherwise run the interrupt handler, if configured, and not in-progress
-	if (interruptIndex && !interruptStackBase)
+	if (interruptIndex)
 	{
-		if (logoStackIndex < LOGO_STACK_DEPTH-1)
+		//if not arrived
+		if (tofinish_line >= WAYPOINT_PROXIMITY_RADIUS) // not crossed the finish line
 		{
-			logoStackIndex++;
-			logoStack[logoStackIndex].frameType = LOGO_FRAME_TYPE_SUBROUTINE;
-			logoStack[logoStackIndex].arg = 0;
-			logoStack[logoStackIndex].returnInstructionIndex = instructionIndex-1;
-			instructionIndex = interruptIndex+1;
-			interruptStackBase = logoStackIndex;
+			if (!interruptStackBase)   //if not in-progress
+			{
+				if (logoStackIndex < LOGO_STACK_DEPTH-1)
+				{
+					logoStackIndex++;
+					logoStack[logoStackIndex].frameType = LOGO_FRAME_TYPE_SUBROUTINE;
+					logoStack[logoStackIndex].arg = 0;
+					logoStack[logoStackIndex].returnInstructionIndex = instructionIndex-1;
+					instructionIndex = interruptIndex+1;
+					interruptStackBase = logoStackIndex;
+				}
+			}
 			process_instructions();
 			navigate_set_goal_height(turtleLocations[PLANE].z);

```

Comment:
Should this line and the following line be here?  I think they might make more sense running after the interrupt completes.

Reply:
Ben,

These lines where already there. They are run after the interrupt completes, but now also each time after MAX_INSTRUCTIONS_PER_CYCLE was reached with larger interrupt programs
But i'm not completely shure about the purpose of those two lines, because this function is also performed in process_instructions()?

Regards,
Kees.